BSP/Inc/BSP_PWM.h

```diff
@@ -0,0 +1,44 @@
+#ifndef BSP_PWM_H
+#define BSP_PWM_H
+
+#include "stm32f4xx.h"
+#include "stm32f4xx_tim.h"
+#include "stm32f4xx_gpio.h"
+#include "stm32f4xx_rcc.h"
+#include "common.h"
+
+
+#define PWM_PERIOD 4000
+
+
+/**
+ * @brief   Initialize all the GPIO pins meant for pwm
+ * @param   None
+ * @return  None
+ */
+void BSP_PWM_Init(void);
+
+/**
+ * @brief   Sets a pin's duty cycle
+ * @param   dutyCycle: duty cycle amount between 0 and the period
+ * @param   pin: pin number whose speed should be changed
+ * @return  ErrorStatus
+ */
+ErrorStatus BSP_PWM_Set(uint8_t pin, uint32_t speed);
+
+/**
+ * @brief   Get current duty cycle of a single pin, return -1 if input is invalid
+ * @param   pin Number
+ * @return  Current PWM duty cycle of pin
+ */
+int BSP_PWM_Get(uint8_t pin);
+
+#define C1_PORT GPIOB //TO MAKE THE CONTACTOR CODE WORK, REMOVE ONCE IT HAS A NEW HOME. This is the port for the input pin
```

Comment:
this line should be at top of header file with other #defines.

Reply:
Why not keep the contactor input pin stuff together?

Reply:
It's good form to put all defines at top of file